packages/shadcn-ui/src/exports/Fields/FieldArray.tsx

```diff
@@ -0,0 +1,59 @@
+import { Field, FieldsByKey } from '@tutim/types';
+import { AddButton, DeleteButton } from '../Buttons';
+import { FieldGroup, getFieldsLayout } from '../Forms';
+import { FieldCollapse } from '../Forms/Collapse';
+
+const getRelativeFieldsByKey = (fieldsByKey: FieldsByKey): FieldsByKey => {
```

Comment:
extract to headless library

Reply:
> extract to headless library

But FieldCollapse depends on the shadcn library as well